File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/MessageResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import com.pengrad.telegrambot.model.Message;
import org.springframework.stereotype.Component;

import java.lang.reflect.Parameter;

@Component
public class MessageResolver implements ParameterResolver<Message> {

    @Override
    public boolean supports(Parameter p) {
        return p.getType().equals(Message.class);
    }

    @Override
    public Message resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.update().message();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/MessageResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import com.pengrad.telegrambot.model.Message
import org.springframework.stereotype.Component
import java.lang.reflect.Parameter

@Component
class MessageResolver : ParameterResolver<Message?> {
    override fun supports(p: Parameter?): Boolean? = p?.getType() == Message::class.java

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): Message? = ctx.update?.message()
}
